.github/scripts/mutate_check_action_pins.py

```
@@ -0,0 +1,205 @@
+            print(f"{'PATTERN MISSING':17}{label}")
+            survivors.append(label)
+            continue
+
+        workspace = Path(tempfile.mkdtemp(prefix="mutant-"))
+        try:
+            (workspace / "scripts").mkdir()
+            (workspace / "scripts" / GUARD.name).write_text(
+                guard.replace(original, broken, 1), encoding="utf-8"
+            )
+            shutil.copy(SUITE, workspace / "scripts" / SUITE.name)
+            result = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)